onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -93,67 +111,39 @@ inline void CopyPrim(uint8_t* target, const uint8_t* source) {
   *reinterpret_cast<T*>(target) = *reinterpret_cast<const T*>(source);
 }
 
+// The function does not check num_axes > 0 but this is expected.
+template <class T>
+static void TypedDoTransposeEltWise(int64_t num_axes, const std::vector<int64_t>& target_dims, size_t num_blocks,
+                                    const std::vector<size_t>& stride, const uint8_t* source, uint8_t* target) {
+  std::vector<MultiIndex> mindex(num_axes);
```

Comment:
If this took args for the end of the source and target data, could we do an explicit check we haven't gone out of bounds? Would it be possible to just do that check once at the end of the 'for' loop that copies?

Reply:
I need to measure how much time would be spent in the testing before answering. Not sure checking at the end of the loop would work. The loop could go outside then go back inside. In the previous version, the pointer stayed all the time within boundaries. It created a failure because instead of writing 1 byte, it was writing 8 bytes at a time: 1 byte inside, 7 bytes outside for the last affectation. It did not fail on my machine because the compilation aligned the pointer to 8 bytes and the pointer on the target buffer always increases.

Reply:
I don't quite follow that. the sizes of the blocks to write should be based on the input dimensions, so how do we get a mismatch where it's attempting to write 8 bytes for a 1 byte block? that sounds more like a bug figuring out block sizes and the check should be where we determine that.

Reply:
I went back and looked at the old code and it seemed the bug was hardcoding a type in this call that had no relation to the data type being copied.
So this:
`CopyPrim<uint64_t>(target, local_source);`
is now
`CopyPrim<T>(target, local_source);`

Is that correct?

If so there was never an incorrect calculation of an offset that violated the bounds was there? It was just that the copy wasn't limited to the correct number of bytes.

Reply:
That's correct. I replaced it by a template to reduce the duplication of code.

Reply:
Based on that I'm not convinced we need to go all out to validate the bounds. There was never an issue with those - it was simply the use of the wrongly sized type in the copy. 

Instead is it possible to craft a unit test that broke with the incorrect copy size without explicitly checking the bounds? you're writing 8 bytes in places where 1 should be written, so I would expect with carefully chosen input shape and transpose perms you could cause a later write to overwrite some of an earlier one leading to the output not matching the expected values. that would prevent regression without needing extra ORT_ENFORCE statements in the loop or exposing internal helper methods just for a unit test.